packages/smart-contracts/test/contracts/BatchEthereumPayments.test.ts

```diff
@@ -0,0 +1,210 @@
+  const provider = new ethers.providers.JsonRpcProvider(networkConfig.url);
+
+  let amount = 100;
+  let feeAmount = 1;
+  let nbTxs: number;
+  let tx;
+
+  before(async () => {
+    [, payeeOne, payeeTwo, feeAddress] = (await ethers.getSigners()).map((s) => s.address);
+    [owner] = await ethers.getSigners();
+    ownerAddress = await owner.getAddress();
+
+    ethFeeProxy = ethereumFeeProxyArtifact.connect(network.name, owner);
+    let ethFeeProxyAddress = ethFeeProxy.address;
+
+    batch = batchPaymentsArtifact.connect(network.name, owner);
+    batchAddress = batch.address;
+
+    if (logInfos) {
+      console.log('ethFeeProxyAddress addr', ethFeeProxyAddress);
+      console.log('batch addr', batchAddress);
+      console.log(
+        'batch Erc20 proxy addr',
+        await batch.paymentErc20FeeProxy.call({ from: ownerAddress }),
+      );
+      console.log(
+        'batch Eth proxy addr',
+        await batch.paymentEthereumFeeProxy.call({ from: ownerAddress }),
+      );
+    }
+  });
+
+  it('Should pay 2 payments and return the excess of ethers', async function () {
+    beforeEthBalance = await provider.getBalance(payeeOne);
+    let totalAmout = BigNumber.from('1000');
+
+    await expect(
+      batch
+        .connect(owner)
+        .batchEthereumPaymentsWithReferenceAndFee(
+          [payeeOne, payeeTwo],
+          [20, 30],
+          [referenceExample1, referenceExample2],
+          [1, 2],
+          feeAddress,
+          {
+            value: totalAmout,
+          },
+        ),
+    )
+      .to.emit(ethFeeProxy, 'TransferWithReferenceAndFee')
+      .withArgs(payeeOne, '20', ethers.utils.keccak256(referenceExample1), '1', feeAddress)
+      .to.emit(ethFeeProxy, 'TransferWithReferenceAndFee')
+      .withArgs(payeeTwo, '30', ethers.utils.keccak256(referenceExample2), '2', feeAddress);
+
+    afterEthBalance = await provider.getBalance(payeeOne);
+    expect(afterEthBalance).to.be.equal(beforeEthBalance.add(20));
+    expect(await provider.getBalance(batchAddress)).to.be.equal(0);
+  });
+
+  it('Should pay 2 payments with the exact amount', async function () {
+    beforeEthBalance = await provider.getBalance(payeeOne);
+    beforeEthBalance1 = await provider.getBalance(payeeTwo);
+    let totalAmout = BigNumber.from('53');
+
+    tx = await batch
+      .connect(owner)
+      .batchEthereumPaymentsWithReferenceAndFee(
+        [payeeOne, payeeTwo],
+        [20, 30],
+        [referenceExample1, referenceExample2],
+        [1, 2],
+        feeAddress,
+        {
+          value: totalAmout,
+        },
+      );
+    await tx.wait();
+
+    afterEthBalance = await provider.getBalance(payeeOne);
+    expect(afterEthBalance).to.be.equal(beforeEthBalance.add(20));
+
+    afterEthBalance1 = await provider.getBalance(payeeTwo);
+    expect(afterEthBalance1).to.be.equal(beforeEthBalance1.add(30));
+  });
+
+  it('Gas evaluation: Should pay multiple Ethereum payments', async function () {
```

Comment:
this test seems to be about gas, but there is nothing in it specific to gas 🤔 (except a log)